tidy3d/components/medium.py

```diff
@@ -1267,6 +1336,257 @@ def elements(self) -> Dict[str, IsotropicMediumType]:
         return dict(xx=self.xx, yy=self.yy, zz=self.zz)
 
 
+class FullyAnisotropicMedium(AbstractMedium):
+    """Fully anisotropic medium.
+
+    Example
+    -------
+    >>> perm = [[2, 0, 0], [0, 1, 0], [0, 0, 3]]
+    >>> cond = [[0.1, 0, 0], [0, 0, 0], [0, 0, 0]]
+    >>> anisotropic_dielectric = FullyAnisotropicMedium(
+    >>>    permittivity=perm, conductivity=cond
+    >>>)
+    """
+
+    permittivity: RealTensor = pd.Field(
+        [[1, 0, 0], [0, 1, 0], [0, 0, 1]],
+        title="Permittivity",
+        description="Relative permittivity tensor.",
+        units=PERMITTIVITY,
+    )
+
+    conductivity: RealTensor = pd.Field(
+        [[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+        title="Conductivity",
+        description="Electric conductivity tensor. Defined such that the imaginary part "
+        "of the complex permittivity at angular frequency omega is given by conductivity/omega.",
+        units=CONDUCTIVITY,
+    )
+
+    @pd.validator("permittivity", always=True)
+    def permittivity_spd_and_ge_one(cls, val):
+        """Check that provided permittivity tensor is symmetric positive definite
+        with eigenvalues >= 1.
+        """
+
+        if not np.allclose(val, np.transpose(val), atol=1.0e-10):
+            raise ValidationError("Provided permittivity tensor is not symmetric.")
+
+        if np.any(np.linalg.eigvals(val) < 1 - 1.0e-10):
+            raise ValidationError("Main diagonal of provided permittivity tensor is not >= 1.")
+
+        return val
+
+    @pd.validator("conductivity", always=True)
+    def conductivity_ge_zero_and_commutes(cls, val, values):
+        """Check that the symmetric part of conductivity tensor commutes with permittivity tensor
+        (that is, simultaneously diagonalizable) with eigenvalues >= 0.
+        """
+
+        perm = values.get("permittivity")
+        cond_sym = 0.5 * (val + np.transpose(val))
+
+        if not np.allclose(
+            np.abs(np.matmul(perm, cond_sym) - np.matmul(cond_sym, perm)), 0, atol=1.0e-10
+        ):
+            raise ValidationError(
+                "Main directions of conductivity and permittivity tensor do not coincide."
+            )
+
+        if np.any(np.linalg.eigvals(cond_sym) < -fp_eps):
+            raise ValidationError("Main diagonal of provided conductivity tensor is not >= 0.")
+
+        return val
+
+    @classmethod
+    def from_diagonal(cls, xx: Medium, yy: Medium, zz: Medium, rotation: RotationType):
+        """Construct a fully anisotropic medium by rotating a diagonally ansisotropic medium.
+
+        Parameters
+        ----------
+        xx : Medium
```

Comment:
```
:class:`.Medium`
```

Reply:
same for RotationType if it will render properly in the docs (ie has a reference in api.rst). Otherwise we'll need to probably be more specific about what a RotationType is here.

Reply:
fixed